service/src/main/java/uk/gov/hmcts/reform/fpl/exceptions/GrantCaseAccessException.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.reform.fpl.exceptions;
+
+import lombok.EqualsAndHashCode;
+import uk.gov.hmcts.reform.fpl.enums.CaseRole;
+
+import java.util.Set;
+
+@EqualsAndHashCode(callSuper = false)
+public class GrantCaseAccessException extends RuntimeException {
+
+    private String caseId;
+    private String localAuthority;
+    private Set<String> userIds;
+    private Set<CaseRole> caseRoles;
+
+    public GrantCaseAccessException(String caseId, Set<String> userIds, Set<CaseRole> caseRoles) {
```

Comment:
I assumed that it is not needed. Second constructor have it as list of users is not know. 
From method perspective:

public void grantAccessToUser(String caseId, String user, Set<CaseRole> roles)

it is not related to any la, can be treated as generic method to give arbitrary roles to any user

Reply:
I see, this scenario is interesting as here we always know LA, was thinking maybe it would be good for support perspective, but let's leave it for now.